tests/app/questionnaire/test_navigation.py

```diff
@@ -719,3 +749,201 @@ def test_build_navigation_confirmation_link_visible_when_all_sections_complete(s
+            Location('multiple-questions-group', 0, 'household-composition'),
+            Location('repeating-group', 0, 'repeating-block-1'),
+            Location('repeating-group', 0, 'repeating-block-2'),
+            Location('extra-cover', 0, 'extra-cover-block'),
+            Location('extra-cover', 0, 'extra-cover-interstitial'),
+            Location('payment-details', 0, 'credit-card'),
+            Location('payment-details', 0, 'expiry-date'),
+            Location('payment-details', 0, 'security-code'),
+            Location('payment-details', 0, 'security-code-interstitial'),
+            Location('extra-cover-items-group', 0, 'extra-cover-items'),
+        ]
+
+        navigation = Navigation(survey, AnswerStore(), metadata, completed_blocks)
+
+        confirmation_link = {
+            'link_name': 'Submit answers',
+            'highlight': False,
+            'repeating': False,
+            'completed': False,
+            'link_url': Location('confirmation-group', 0, 'confirmation').url(metadata)
+        }
+
+        self.assertIn(confirmation_link, navigation.build_navigation('property-details', 0))
+
+    def test_build_navigation_summary_link_not_visible_for_survey_with_confirmation(self):
+        survey = load_schema_file("test_navigation_confirmation.json")
+        metadata = {
+            "eq_id": '1',
+            "collection_exercise_sid": '999',
+            "form_type": "some_form"
+        }
+
+        completed_blocks = [
+            Location('property-details', 0, 'insurance-type'),
+            Location('property-details', 0, 'insurance-address'),
+            Location('property-details', 0, 'property-interstitial'),
+            Location('house-details', 0, 'house-type'),
+            Location('multiple-questions-group', 0, 'household-composition'),
+            Location('repeating-group', 0, 'repeating-block-1'),
+            Location('repeating-group', 0, 'repeating-block-2'),
+            Location('extra-cover', 0, 'extra-cover-block'),
+            Location('extra-cover', 0, 'extra-cover-interstitial'),
+            Location('payment-details', 0, 'credit-card'),
+            Location('payment-details', 0, 'expiry-date'),
+            Location('payment-details', 0, 'security-code'),
+            Location('payment-details', 0, 'security-code-interstitial'),
+            Location('extra-cover-items-group', 0, 'extra-cover-items'),
+        ]
+
+        navigation = Navigation(survey, AnswerStore(), metadata, completed_blocks)
+
+        confirmation_link = {
+            'link_name': 'Summary',
+            'highlight': False,
+            'repeating': False,
+            'completed': False,
+            'link_url': Location('summary-group', 0, 'summary').url(metadata)
+        }
+
+        self.assertNotIn(confirmation_link, navigation.build_navigation('property-details', 0))
+
+    @pytest.mark.xfail(reason="Summary should not display since there is a hidden group that isn't completed")
```

Comment:
This should be fixed with my latest changes. I've removed the xFail annotation, will be pushing up changes soon.